ContentPatcher/Framework/Conditions/TokenString.cs

```diff
@@ -110,7 +110,10 @@ public TokenString(IEnumerable<ILexToken> lexTokens, IContext context, LogPathBu
                     tokensUsed ??= new();
                     tokensUsed.Add(token.Name);
 
-                    isMutable = isMutable || token.IsMutable;
+                    if (!token.RequiresInput || !token.IsDeterministicForInput)
+                    {
+                        isMutable = isMutable || token.IsMutable;
+                    }
```

Comment:
Unfortunately we can't ignore token mutability just because it doesn't require input. For example, `{{Time}}` doesn't have input but still changes its values every seven seconds or so. (The token can also have optional input even if `RequiresInput` is false.)

I also suggest adding a comment to explain the optimization here.

```suggestion
                    if (!token.IsDeterministicForInput)
                    {
                        // If every token is deterministic, then the resulting string must be immutable. If this token
                        // is deterministic but it receives non-deterministic input, the string will be marked mutable
                        // when we recursively check those tokens.
                        isMutable = isMutable || token.IsMutable;
                    }
```